core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -71,6 +82,14 @@ protected function prepare($objPage)
 		// Static URLs
 		$this->setStaticUrls();
 
+		// Set response context
+		/** @var WebpageContext $responseContext */
+		$responseContext = $container->get(ResponseContextFactory::class)->createAndSetCurrent(WebpageContext::class);
```

Comment:
isn't that a bit late and should be done in the controller?

Reply:
It's exactly where it's supposed to be. In the respective page type at the very beginning?

Reply:
The actual controller is here: https://github.com/contao/contao/blob/4.x/core-bundle/src/Resources/contao/controllers/FrontendIndex.php#L88
this one is used by Symfony. However I agree `PageRegular` is _kind of_ a sub-controller for each page type. But I'm pretty sure there are use-cases to create your own instance of `PageRegular`, and that should never override the response context imho?

Reply:
Yeah where we set the context in this PR already, I just did it in PageRegular again. But I don't really care how we do that here TBH because it's going to be different with your PR anyway.